### PR TITLE
Calc Solution

### PR DESCRIPTION
**Pull request by vuyko23**

### CODE REVIEW

File: src/test/java/core/basesyntax/CalculatorImplTest.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        double actual = calculator.calculate(2.22, 3.33, PLUS);
+        double expected = 5.55;
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void addTwoNegativeOperand_Ok() {
+        double actual = calculator.calculate(-2.1, -6.23, PLUS);
+        double expected = -8.33;
+        assertEquals(expected, actual,DELTA);
```
Comment (DanyloIT) [inline src/test/java/core/basesyntax/CalculatorImplTest.java:34]: add space before DELTA)

**Review by DanyloIT — APPROVED** [review]

File: src/main/java/core/basesyntax/CalculatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public double calculate(double a, double b, char operand) {
+        switch (operand) {
+            case '+':
+                return a + b;
+            case '-':
+                return a - b;
+            case '*':
+                if (a == 0 || b == 0) {
+                    return 0;
+                }
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/CalculatorImpl.java:14]: no need. it won't be any faster

File: src/test/java/core/basesyntax/CalculatorImplTest.java (hunk 1 above)
Comment (Max-Pochepets) [reply src/test/java/core/basesyntax/CalculatorImplTest.java:34]: agree

File: src/test/java/core/basesyntax/CalculatorImplTest.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    private static final double DELTA = 0.0001;
+    private static Calculator calculator;
+
+    @BeforeAll
+    static void beforeAll() {
+        calculator = new CalculatorImpl();
+    }
+
+    @Test
+    void addTwoPositiveOperand_Ok() {
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorImplTest.java:24]: what method do we test?

File: src/test/java/core/basesyntax/CalculatorImplTest.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    @Test
+    void addNegativeToZero_Ok() {
+        double actual = calculator.calculate(0, -7.77, PLUS);
+        double expected = -7.77;
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void addToMaxDouble() {
+        double actual = calculator.calculate(Double.MAX_VALUE, Double.MAX_VALUE, PLUS);
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorImplTest.java:81]: what about mix of positive and negative edge values?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/CalculatorImplTest.java [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    private static final double DELTA = 0.0001;
+    private static Calculator calculator;
+
+    @BeforeAll
+    static void beforeAll() {
+        calculator = new CalculatorImpl();
+    }
+
+    @Test
+    void addTwoPositiveValues_Ok() {
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorImplTest.java:24]: what about this [comment](https://github.com/mate-academy/jv-lambda-calc/pull/181#discussion_r617414213)? [useful link](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names)

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]
